pkg/x/sylt/workflow.go

```diff
@@ -9,18 +9,49 @@ import (
 
 var ErrDuplicateAction = errors.New("duplicate action name and type")
 
-// Workflow deploys OpenTofu stacks and Kubernetes apps.
+func NewWorkflow(opts ...WorkflowOption) *Workflow {
+	opt := defaultWorkflowOpts()
+	for _, o := range opts {
+		o(&opt)
+	}
+	return &Workflow{
+		opts: opt,
+	}
+}
+
+// Workflow runs actions implementing the [Actioner] interface, to perform
+// things like deploying terra stacks and kube apps.
+// Use the [NewWorkflow] function to create a new workflow.
 type Workflow struct {
-	Opts    WorkflowOpts
+	opts    workflowOpts
 	actions []Actioner
 	mu      sync.Mutex
 }
 
-type WorkflowOpts struct {
+type WorkflowOption func(*workflowOpts)
+
+// WithWorkflowDryRun sets the dry run option for Workflow.
+func WithWorkflowDryRun(b bool) WorkflowOption {
```

Comment:
bool is already false, no need to pass it unless it is `true`

Reply:
I like to pass bools to these, because when you go to write your CLI (or whatever will use it), it is much cleaner to just pass your bool, rather than doing an optional append of options... E.g.

```go
opts := []SomeOption{}
if dryRun {
  opts = append(opts, WithDryRun())
}
```

vs

```go
opts := []SomeOption{WithDryRun(dryRun)}
```

I've had to do this enough that I just prefer it this way 🤷‍♂️